add support for token signin

jakzo opened the pull request:
I'm looking to add Home Assistant support to my dehumidifier and reverse engineering the app shows it uses Ayla Networks but in the app you log into De'Longhi which generates a token and calls this token_sign_in endpoint I've added.

I tried to make the changes backwards-compatible. Tested these changes using the Europe URL and both sync and async token signin work. I'm planning to open a PR later to add constants for the Dehumidifier once I've figured them all out.

rewardone commented:
While this technically works, I'm not particularly satisfied with this implementation.

1. Define what the token is. ._token is ambiguous. If it is an access_token, use the ._access_token property. These terms are generally understood to be JWT tokens and their roles are clear. If the De'Longhi token is not a JWT ... there should be sufficient checking during the sso_sign_in that the token's length, charset, format, etc. match what is uniquely expected. 

2. Instead of modifying the sign_in functions, since this is a new endpoint with a new type of login flow. [Ayla](https://docs.aylanetworks.com/reference/login) defines this as an SSO endpoint, so sso_sign_in would be the most appropriate. 

3. I'm not sure if the De'Longhi app or you would be required to call the [Token Validation](https://docs.aylanetworks.com/reference/token-validation) function. I would assume that De'Longhi handles all authentication and finishes with an authenticated session to Ayla. 

4. Regarding constants, I also mentioned to the Fujitsu contributor that Ayla should not be responsible for device specific constants. Recommend breaking out a separate library to handle De'Longhi specific things. (see [Culligan](https://github.com/rewardone/Culligan) as an example).

jakzo commented:
I split out the sso signin functions and renamed the token to `sso_token`. Hope that is a bit clearer but let me know if you think it should be changed.

rewardone commented:
You are a blessing, thank you for the PR and documentation!
